### PR TITLE
PHOENIX-7430 Remove commons-collections dependency in queryserver

### PR DESCRIPTION
**Pull request by NihalJain**
- Also remove imports and replace with equivalent code

### CODE REVIEW

File: phoenix-queryserver/src/main/java/org/apache/phoenix/util/InstanceResolver.java [hunk 1]
```diff
@@ -68,10 +67,13 @@ public static <T> T getSingleton(Class<T> clazz, T defaultInstance) {
     @SuppressWarnings("unchecked")
     public static <T> List get(Class<T> clazz, List<T> defaultInstances) {
         Iterator<T> iterator = ServiceLoader.load(clazz).iterator();
+        List<T> instances = new ArrayList<>();
```
Comment (NihalJain) [inline phoenix-queryserver/src/main/java/org/apache/phoenix/util/InstanceResolver.java:70]: There is no guava in PQS, so had to do this inplace using lambdas. Should we sync same logic in Phoenix? This class seems to be present at both places.

File: pom.xml [hunk 2]
```diff
@@ -723,12 +722,6 @@
                   </exclusion>
                 </exclusions>
             </dependency>
```
Comment (NihalJain) [inline pom.xml:724]: We do not have restrict plugin in PQS. Should we add one here and sync all restricts or just add commons-collections restricts?

**Comment by NihalJain** [comment]
Hi @stoty, could you please review?

File: pom.xml (hunk 2 above)
Comment (stoty) [reply pom.xml:724]: It's up to you.
PQS has minimal code.
If you decide to add the restricions, then I think it's better to sync everything from the main repo.

**Review by stoty — APPROVED** [review]
+1 LGTM

File: phoenix-queryserver/src/main/java/org/apache/phoenix/util/InstanceResolver.java (hunk 1 above)
Comment (NihalJain) [reply phoenix-queryserver/src/main/java/org/apache/phoenix/util/InstanceResolver.java:70]: sure makes sense

File: pom.xml (hunk 2 above)
Comment (NihalJain) [reply pom.xml:724]: Yes make sense, let me raise another JIRA for same, this will keep changes in PR as per JIRA title

Comment (NihalJain) [reply pom.xml:724]: Raised PHOENIX-7438

**Pull request merged**